medusa/providers/generic_provider.py

```diff
@@ -234,139 +234,166 @@ def find_search_results(self, show, episodes, search_mode, forced_search=False,
+                                                         ).parse(search_result.name)
             except (InvalidNameException, InvalidShowException) as error:
                 logger.log(u"{error}".format(error=error), logger.DEBUG)
+                search_result.add_cache_entry = False
+                search_result.result_wanted = False
                 continue
 
-            show_object = parse_result.show
-            quality = parse_result.quality
-            release_group = parse_result.release_group
-            version = parse_result.version
-            add_cache_entry = False
+            # I don't know why i'm doing this. Maybe remove it later on all together, now i've added the parsed_result
+            # to the search_result.
+            search_result.show = search_result.parsed_result.show
+            search_result.quality = search_result.parsed_result.quality
+            search_result.release_group = search_result.parsed_result.release_group
+            search_result.version = search_result.parsed_result.version
+            search_result.actual_season = search_result.parsed_result.season_number
+            search_result.actual_episodes = search_result.parsed_result.episode_numbers
 
             if not manual_search:
-                if not (show_object.air_by_date or show_object.sports):
+                if not (search_result.show.air_by_date or search_result.show.sports):
                     if search_mode == 'sponly':
-                        if parse_result.episode_numbers:
+                        if search_result.parsed_result.episode_numbers:
                             logger.log(
                                 'This is supposed to be a season pack search but the result %s is not a valid '
-                                'season pack, skipping it' % title, logger.DEBUG
+                                'season pack, skipping it' % search_result.name, logger.DEBUG
                             )
-                            add_cache_entry = True
+                            search_result.result_wanted = False
+                            continue
                         elif not [ep for ep in episodes if
-                                  parse_result.season_number == (ep.season, ep.scene_season)[ep.show.is_scene]]:
+                                  search_result.parsed_result.season_number == (ep.season, ep.scene_season)
+                                  [ep.show.is_scene]]:
                             logger.log(
-                                'This season result %s is for a season we are not searching for, skipping it' % title,
+                                'This season result %s is for a season we are not searching for, '
+                                'skipping it' % search_result.name,
                                 logger.DEBUG
                             )
-                            add_cache_entry = True
-
```

Comment:
It's an opt-out. The add_cache_entry is set to True by default. Then only under rare conditions (mostly exceptions) it's toggled to False.
https://github.com/pymedusa/Medusa/pull/1997/files#diff-90b4db298132f61ff3ce451582cd4a63R108